core/shared/src/main/scala/zio/ZLayer.scala

```diff
@@ -474,6 +474,181 @@ object ZLayer extends ZLayerCompanionVersionSpecific {
   ): ZLayer[RIn, E, ROut] =
     ZLayer.fromZIO(zio)
 
+  object Derive {
+
+    /**
```

Comment:
Documentation could be updated here to reflect the changes we have made.

Reply:
Updated Scaladoc.